# BERDO Analysis

## 1. Executive Summary

This analysis examines Boston’s 2025 BERDO reporting dataset to identify patterns in reporting completeness, property classification, energy performance, and potential compliance risk. The findings suggest that BERDO implementation should be understood as both a reporting challenge and a long-term building performance challenge.

| Key Metric | Finding | Interpretation |
|---|---:|---|
| Buildings listed as In Compliance | 64.0% | Most buildings in the dataset have submitted required reporting information. |
| Buildings listed as Not Submitted | 28.3% | A substantial share of buildings remain outside the visible reporting system. |
| Missing Site EUI among Not Submitted buildings | 34.1% | Missing energy data is concentrated among buildings with incomplete reporting. |

The most important insight is that buildings with missing property type or missing Site EUI are more difficult to evaluate, prioritize, and support. These gaps may reflect deeper capacity, ownership, or coordination barriers.

The analysis identifies two support needs. One group includes buildings that may need reporting assistance, outreach, and ownership coordination support. The second includes buildings with complete data but higher energy intensity, which may need deeper technical review, retrofit planning, and resource navigation.

The analysis also calculates estimated GHG emissions intensity in kg CO₂e per square foot to connect the screening framework more directly to BERDO’s official emissions performance metric.

### 1.1 Tools Used

- Python
  
- pandas
 
- matplotlib

- Excel


### 1.2 Project Objective

This project analyzes BERDO compliance data for Boston buildings to evaluate how energy performance, ownership structure, and building characteristics relate to compliance outcomes.

The goal is to identify where emissions reduction efforts can create the greatest practical impact by distinguishing between buildings that need reporting support and buildings that require deeper retrofit intervention.

Rather than treating all buildings equally, this analysis prioritizes buildings based on energy intensity, operational complexity, and likelihood of compliance barriers.


## 2. Methodology

### 2.1 Data Preparation

The analysis uses the 2025 BERDO-reported Energy and Water Metrics dataset for Boston buildings.

Data was imported and cleaned using Python and pandas to improve consistency and analytical usability. Key preprocessing steps included:

1. Standardizing column names for easier querying and analysis.

2. Renaming major variables such as property type, Site EUI, emissions, and compliance status for clarity.

3. Removing records with missing Site EUI values when energy performance comparisons required valid observations.

4. Preserving the full dataset separately for reporting gap and missing data analysis

This allowed the project to distinguish between operational performance analysis and reporting completeness analysis.


### 2.2 Core Variables

The analysis focused on variables most directly connected to BERDO compliance and emissions performance:

- Property type

- Site Energy Use Intensity (Site EUI)

- Total site energy usage

- Greenhouse gas emissions

- Compliance status

- Building size (gross floor area)

- Ownership patterns inferred from owner name classification

These variables were selected to evaluate both energy performance and implementation difficulty.


### 2.3 Analytical Framework

The project combines descriptive analysis, comparative metrics, and rule-based prioritization to identify high-impact intervention groups.

Key methods included:

- Compliance distribution analysis across the full dataset

- Property type and ownership gap analysis for missing records

- Grouped comparison of average Site EUI across compliance categories

- Correlation analysis between building size and energy intensity

- Complexity classification based on likely retrofit and ownership challenges

- Priority segmentation using both energy intensity and implementation complexity

This framework supports a more practical interpretation of BERDO compliance by connecting emissions performance to real implementation barriers.


### 2.4 Method Notes and Limitations

The ownership categories used in this analysis are interpretive and based on patterns in owner names. They are not official BERDO classifications.

The analysis distinguishes between reporting compliance and emissions performance. A building listed as In Compliance has submitted required reporting information, but this does not necessarily mean the building has achieved long-term emissions reduction targets.

Site EUI comparisons exclude records with missing Site EUI values, while missing data patterns are analyzed using the full dataset.


## 3. Key Insights

### 3.1 Missing Data as a Signal of Compliance Gaps

1,579 buildings across the full dataset fall into the not submitted category. Notably, buildings with missing property type are overwhelmingly concentrated within this group — 1,579 of 1,653 (95.5%) buildings missing a property type are also not submitted, suggesting these two data gaps occur together rather than independently.

### 3.2 BERDO has two intervention groups

- Buildings with missing or incomplete reporting data need outreach, reporting help, and owner coordination.

- Buildings with high Site EUI need deeper energy reduction planning, retrofit support, financing, or technical assistance.

Multifamily is the largest category, but not the biggest reporting challenge. Multifamily Housing has more than 2,000 buildings, making it the largest category in the dataset. But the biggest reporting gap is concentrated in the No Property Type group, not multifamily.

### 3.3 Ownership complexity may help explain reporting gaps

Buildings missing property type are spread across ownership patterns such as LLCs, public sector entities, condominiums, individuals, trusts, and corporate owners. This suggests that reporting gaps may be tied to administrative complexity, fragmented ownership, or limited reporting capacity.

### 3.4 Energy intensity varies more by building function than by building size

Correlation between gross floor area and Site EUI is around 0.099. That means larger buildings are not automatically more energy intensive per square foot.

### 3.5 Specialized buildings show the highest Site EUI

Restaurants, fast food restaurants, healthcare facilities, laboratories, supermarkets, data centers, and industrial uses show some of the highest average Site EUI values. These buildings likely need tailored energy strategies because their energy use is tied to specialized operations.

### 3.6 State pathway buildings have higher average Site EUI

Buildings in the State category show a higher average Site EUI than buildings marked In Compliance. This does not prove causation, but it suggests that some compliance pathways include more energy-intensive buildings.


## 4. Interpretation

These findings show that BERDO compliance challenges are not only about energy performance. They are also about reporting capacity, ownership complexity, and data completeness.

Missing property type is connected to non-submission, suggesting that missing data can serve as an early warning signal for compliance risk. Buildings with incomplete reporting likely need outreach and reporting support, while buildings with high Site EUI may need deeper retrofit planning, technical assistance, or financing.

My analysis supports a phased approach: help incomplete reporters first, and prioritize high-energy buildings for long-term emissions reduction planning.


## 5. Stakeholder Alignment

This pattern aligns with stakeholder interviews conducted as part of a BERDO case study.

Common barriers included difficulty identifying building owners, limited technical capacity among smaller and fragmented ownership groups, complex and time-intensive reporting processes, and limited access to engineering expertise and retrofit planning support.

These factors help explain why some buildings submit incomplete data or fail to report entirely.


## 6. Why This Matters

This finding reframes missing data from a limitation into a diagnostic tool.
Missing property type can serve as an early warning signal for buildings at greater risk of noncompliance.

This creates an opportunity to:

- Target outreach and technical assistance more effectively.

- Prioritize buildings with the highest likelihood of reporting gaps.

- Improve compliance through proactive intervention rather than reactive enforcement.


## 7. Final Takeaway

Data completeness and compliance are deeply connected.

In this dataset, missing information is not noise, it is a clear and measurable signal of where the system is breaking down.

There are two different intervention groups:

**Group 1: Not Submitted → needs outreach + reporting support**

**Group 2: High EUI → needs retrofit + energy reduction**

## 8. Data Setup

### 8.1 Libraries

In [1]:
#Load libraries
import pandas as pd
import os
import matplotlib.pyplot as plt
import numpy as np
from pathlib import Path

### 8.2 File Path

In [2]:
file_path = Path.home() / "Documents" / "BERDO Analysis" / "2025-reported-energy-and-water-metrics.xlsx"

#Check if the file exists
if not file_path.exists():
    raise FileNotFoundError("Dataset not found. Please check the file path.")

### 8.3 Load Dataset

In [3]:
def load_dataset(filepath: Path) -> pd.DataFrame:
    """
    Load BERDO dataset with error handling.

    Parameters
    ----------
    filepath : Path
        Path to the Excel file.

    Returns
    -------
    pd.DataFrame
        Loaded dataset.

    Raises
    ------
    FileNotFoundError
        If the file does not exist.
    ValueError
        If the file cannot be read.
    """
    if not filepath.exists():
        raise FileNotFoundError(
            f"Dataset not found at {filepath}. Please check the file path."
        )

    try:
        df = pd.read_excel(filepath, header=0)
        print(f"✓ Dataset loaded successfully: {df.shape[0]:,} rows, {df.shape[1]} columns")
        return df
    except Exception as e:
        raise ValueError(f"Error reading Excel file: {e}")

#Load dataset
df = load_dataset(file_path)

✓ Dataset loaded successfully: 5,580 rows, 54 columns


# Analysis

## 9. Column Renaming and Data Cleaning

### 9.1 Rename Columns and Filter Analysis Dataset

In [7]:
#Define reusable table styles

TABLE_STYLES = [
    {
        "selector": "caption",
        "props": [
            ("caption-side", "top"),
            ("font-size", "18px"),
            ("font-weight", "bold"),
            ("padding", "10px")
        ]
    },
    {
        "selector": "th",
        "props": [
            ("text-align", "center"),
            ("font-size", "14px"),
            ("font-weight", "bold"),
            ("padding", "10px"),
            ("border", "1px solid #DADADA")
        ]
    },
    {
        "selector": "td",
        "props": [
            ("text-align", "center"),
            ("padding", "10px"),
            ("font-size", "13px"),
            ("border", "1px solid #DADADA")
        ]
    }
]

#Define reusable table styling function

def style_table(styler, caption, format_dict=None):
    styled = (
        styler
        .hide(axis="index")
        .set_caption(caption)
        .set_properties(**{
            "text-align": "center",
            "padding": "10px",
            "font-size": "13px",
            "border": "1px solid #DADADA"
        })
        .set_table_styles(TABLE_STYLES)
    )

    if format_dict is not None:
        styled = styled.format(format_dict)

    return styled

In [8]:
#Define expected source columns

required_columns = [
    "Largest Property Type",
    "Reported Gross Floor Area (Sq Ft)",
    "Site EUI (Energy Use Intensity kBtu/ft²)",
    "Total Site Energy Usage (kBtu)",
    "Estimated Total GHG Emissions e(kgCO2e)",
    "Natural Gas Usage (kBtu)",
    "Electricity Usage (kWh)",
    "Reporting Compliance Status",
    "First Emissions Compliance Year (Projected)",
    "Building Address",
    "Property Owner Name"
]

#Check for missing expected columns

missing_columns = [
    col for col in required_columns
    if col not in df.columns
]

if missing_columns:
    raise KeyError(
        f"Missing expected columns: {missing_columns}"
    )

#Rename important columns for easier analysis

df = df.rename(columns={
    "Largest Property Type": "property_type",
    "Reported Gross Floor Area (Sq Ft)": "gross_floor_area",
    "Site EUI (Energy Use Intensity kBtu/ft²)": "site_eui",
    "Total Site Energy Usage (kBtu)": "total_site_energy",
    "Estimated Total GHG Emissions e(kgCO2e)": "ghg_emissions",
    "Natural Gas Usage (kBtu)": "natural_gas_usage",
    "Electricity Usage (kWh)": "electricity_usage",
    "Reporting Compliance Status": "compliance_status",
    "First Emissions Compliance Year (Projected)": "compliance_year"
})

#Convert numeric columns

numeric_columns = [
    "gross_floor_area",
    "site_eui",
    "total_site_energy",
    "ghg_emissions",
    "natural_gas_usage",
    "electricity_usage"
]

for col in numeric_columns:
    df[col] = pd.to_numeric(
        df[col],
        errors="coerce"
    )

#Clean text columns

text_columns = [
    "property_type",
    "compliance_status",
    "Building Address",
    "Property Owner Name"
]

for col in text_columns:
    df[col] = (
        df[col]
        .astype("string")
        .str.strip()
    )

#Create full dataset before filtering

df_full = df.copy()

#Filter dataset for valid Site EUI values only

initial_row_count = df_full.shape[0]

df_clean = (
    df_full
    .dropna(subset=["site_eui"])
    .copy()
)

final_row_count = df_clean.shape[0]

#Calculate removed records

rows_removed = initial_row_count - final_row_count

missing_pct = (
    rows_removed
    / initial_row_count
    * 100
)

#Create data quality summary table

quality_summary = pd.DataFrame({
    "Data Quality Metric": [
        "Initial Records",
        "Records with Valid Site EUI",
        "Records Removed",
        "Percent Removed"
    ],
    "Result": [
        f"{initial_row_count:,}",
        f"{final_row_count:,}",
        f"{rows_removed:,}",
        f"{missing_pct:.1f}%"
    ]
})

#Display data quality summary table

display(
    style_table(
        quality_summary.style,
        "Data Quality Summary"
    )
)

#Preview cleaned dataset

preview_columns = [
    "property_type",
    "gross_floor_area",
    "site_eui",
    "ghg_emissions",
    "compliance_status"
]

display(
    style_table(
        df_clean[preview_columns].head().style,
        "Preview of Cleaned Analysis Dataset",
        {
            "gross_floor_area": "{:,.0f}",
            "site_eui": "{:.2f}",
            "ghg_emissions": "{:,.2f}"
        }
    )
)

Data Quality Metric,Result
Initial Records,"5,580"
Records with Valid Site EUI,"3,678"
Records Removed,"1,902"
Percent Removed,34.1%


property_type,gross_floor_area,site_eui,ghg_emissions,compliance_status
Office,"251,688",70.90,"998,124.32",in compliance
Multifamily Housing,"135,344",55.60,"409,941.74",in compliance
Office,"255,383",45.70,"665,694.65",in compliance
Office,"134,789",56.20,"440,241.97",in compliance
Office,"551,954",49.20,"1,548,997.57",in compliance


### Observation

After filtering for valid Site EUI values, the analysis dataset decreased from 5,580 records to 3,678 records. This removed 1,902 records, or 34.1% of the full dataset, due to missing energy intensity data.

This is important because Site EUI serves as one of the strongest indicators of operational energy performance and potential emissions reduction opportunity. Removing records with missing Site EUI improves the reliability of energy performance comparisons by ensuring that grouped analysis is based only on valid observations.

At the same time, preserving the full dataset separately allows reporting gaps and missing-data patterns to be analyzed independently, preventing compliance failures from being hidden during cleaning. This distinction strengthens the overall analysis by separating operational performance problems from reporting completeness problems.

### 9.2 Features Used
The analysis focuses on variables directly related to compliance and energy performance:
- Property type
- Site Energy Use Intensity (EUI)
- Total energy usage
- Greenhouse gas (GHG) emissions
- Compliance status

## 10. Compliance Distribution

In [9]:
#Create compliance distribution table
compliance_table = (
    df_full["compliance_status"]
    .value_counts(dropna=False)
    .reset_index()
)

#Rename columns
compliance_table.columns = ["Compliance Status", "Building Count"]

#Calculate percentage distribution
compliance_table["Percent of Buildings"] = (
    compliance_table["Building Count"] /
    compliance_table["Building Count"].sum()
)

#Display styled table
display(style_table(
    compliance_table.style,
    "BERDO Compliance Distribution (2025)",
    {"Building Count": "{:,}", "Percent of Buildings": "{:.1%}"}
))

Compliance Status,Building Count,Percent of Buildings
in compliance,"3,569",64.0%
not submitted,"1,579",28.3%
pending revisions,223,4.0%
state,116,2.1%
,93,1.7%


### Observation

The dataset is heavily concentrated in the In Compliance category, with 3,569 buildings (64.0%) listed as having met BERDO reporting requirements. A total of 1,579 buildings fall into the Not Submitted category, representing a significant portion of the dataset. An additional 93 records have no compliance status recorded and are treated as unknown. Smaller groups include Pending Revisions with 223 buildings (4.0%) and State pathway compliance with 116 buildings (2.1%).

This distribution shows that while most buildings submitted required reporting information, a substantial group remains outside the complete reporting system. Based on the broader case study context, these challenges may be linked to fragmented ownership structures, limited technical capacity, or weaker access to engineering and reporting support, particularly among smaller condominium associations, LLC-managed properties, and under-resourced ownership groups.

Rather than representing isolated exceptions, these non-submitted buildings may represent a high-impact intervention group because reporting gaps can overlap with future retrofit planning challenges. This supports the need for targeted outreach, technical assistance, and proactive compliance support rather than relying only on reactive enforcement.

## 11. Property Type Missing Site EUI Analysis

In [ ]:
#property type missing Site EUI analysis

#count all buildings by property type
total_counts = (
    df_full["property_type"]
    .fillna("No Property Type")
    .value_counts()
)

#identify buildings missing Site EUI
missing_eui = df_full[df_full["site_eui"].isna()]

#count missing Site EUI records by property type
missing_eui_counts = (
    missing_eui["property_type"]
    .fillna("No Property Type")
    .value_counts()
)

#combine counts into one table
full_table = pd.DataFrame({
    "Total Buildings": total_counts,
    "Missing Site EUI": missing_eui_counts
}).fillna(0)

#convert columns to integers
full_table["Total Buildings"] = full_table["Total Buildings"].astype(int)
full_table["Missing Site EUI"] = full_table["Missing Site EUI"].astype(int)

#calculate percent missing Site EUI
full_table["Percent Missing Site EUI"] = (
    full_table["Missing Site EUI"] / full_table["Total Buildings"] * 100
).round(1)

#keep top 10 property types by total building count
top10 = full_table.sort_values("Percent Missing Site EUI", ascending=False).head(10)

#group remaining property types into Other
other_counts = full_table.drop(top10.index)[
    ["Total Buildings", "Missing Site EUI"]
].sum()

other_percent = (
    other_counts["Missing Site EUI"] / other_counts["Total Buildings"] * 100
    if other_counts["Total Buildings"] > 0 else 0
)

other_row = pd.Series({
    "Total Buildings": other_counts["Total Buildings"],
    "Missing Site EUI": other_counts["Missing Site EUI"],
    "Percent Missing Site EUI": round(other_percent, 1)
})

#create final table
property_table = pd.concat([top10, pd.DataFrame(other_row).T])
property_table.index = list(top10.index) + ["Other"]

property_table = (
    property_table
    .reset_index()
    .rename(columns={"index": "Property Type"})
)

#display styled table
display(
    property_table.style
    .hide(axis="index")
    .set_caption("Property Types with Highest Missing Site EUI Rates")
    .format({
        "Total Buildings": "{:,}",
        "Missing Site EUI": "{:,}",
        "Percent Missing Site EUI": "{:.1f}%"
    })
    .set_properties(**{
        "text-align": "center",
        "padding": "12px",
        "font-size": "14px",
        "border": "1px solid #DADADA"
    })
    .set_table_styles(TABLE_STYLES)
)

In [ ]:
#Property type missing Site EUI visualization

#Normalized values for color intensity
values = property_table["Percent Missing Site EUI"]
normalized = values / values.max()

#Reds colormap for missing data intensity
colors = plt.cm.Reds(normalized)

#Bar chart
plt.figure(figsize=(12, 6))

plt.bar(
    property_table["Property Type"],
    property_table["Percent Missing Site EUI"],
    color=colors
)

plt.title(
    "Property Types with Highest Missing Site EUI Rates",
    fontsize=14,
    fontweight="bold"
)

plt.xlabel("Property Type", fontsize=12)
plt.ylabel("Percent Missing Site EUI (%)", fontsize=12)

plt.xticks(rotation=45, ha="right")

#Percentage labels above bars
for i, value in enumerate(property_table["Percent Missing Site EUI"]):
    plt.text(
        i,
        value + 0.8,
        f"{value:.1f}%",
        ha="center",
        fontsize=10
    )

plt.tight_layout()
plt.show()

**Observation**

Buildings with missing property type are highly concentrated in records with missing energy performance data. The No Property Type category contains 1,653 total buildings, with 1,652 missing Site EUI, resulting in a 99.9% missing Site EUI rate. By comparison, Multifamily Housing, despite being the largest building category with 2,104 buildings, has only 52 records missing Site EUI.

This shows that missing property classification and missing energy performance data occur together. Missing property type is therefore not simply a data quality issue, but a strong indicator of reporting incompleteness and potential compliance risk.

## 12. Missing Property Type and Compliance Risk

In [ ]:
#Filter buildings with missing property type
missing_property = df_full[df_full["property_type"].isna()].copy()

#Calculate missing property type summary
total_buildings = len(df_full)
missing_property_count = len(missing_property)
missing_property_pct = missing_property_count / total_buildings * 100

missing_property_missing_eui_count = missing_property["site_eui"].isna().sum()
missing_property_missing_eui_pct = (
    missing_property_missing_eui_count / missing_property_count * 100
)

#Create missing property type summary table
missing_property_summary = pd.DataFrame({
    "Metric": [
        "Total Buildings",
        "Buildings Missing Property Type",
        "Percent Missing Property Type",
        "Missing Property Type and Missing Site EUI",
        "Percent of Missing Property Type Group Missing Site EUI"
    ],
    "Result": [
        f"{total_buildings:,}",
        f"{missing_property_count:,}",
        f"{missing_property_pct:.1f}%",
        f"{missing_property_missing_eui_count:,}",
        f"{missing_property_missing_eui_pct:.1f}%"
    ]
})

#Display missing property type summary table
display(
    style_table(
        missing_property_summary.style,
        "Missing Property Type Summary"
    )
)

#Display preview of buildings missing property type
display(
    style_table(
        missing_property[[
            "property_type",
            "gross_floor_area",
            "Building Address",
            "Property Owner Name",
            "site_eui",
            "compliance_status"
        ]].head(20).style,
        "Buildings with Missing Property Type"
    )
)

### Observation

In [ ]:
#Create indicator for whether property type is missing
df_full["missing_property_type"] = df_full["property_type"].isna()

#Compare missing Site EUI rates by property type availability
missing_eui_by_property_status = (
    df_full
    .groupby("missing_property_type")["site_eui"]
    .apply(lambda x: x.isna().mean() * 100)
    .rename("Percent Missing Site EUI")
    .reset_index()
)

#Rename labels for readability
missing_eui_by_property_status["missing_property_type"] = (
    missing_eui_by_property_status["missing_property_type"].replace({
        False: "Property Type Present",
        True: "Property Type Missing"
    })
)

#Rename column
missing_eui_by_property_status = missing_eui_by_property_status.rename(columns={
    "missing_property_type": "Property Type Status"
})

#Display formatted table
display(
    style_table(
        missing_eui_by_property_status.style,
        "Missing Site EUI by Property Type Availability",
        {"Percent Missing Site EUI": "{:.1f}%"}
    )
)

### Observation

Buildings missing property type are far more likely to also be missing Site EUI values. Only 6.4% of buildings with a listed property type are missing Site EUI, compared with 99.9% of buildings without a property type.

This pattern shows that missing property classification and missing energy performance data are not separate issues. They occur together in the dataset and point to a reporting completeness problem.

Missing property type can therefore serve as a practical reporting risk signal. It does not explain why the data is missing, but it helps identify records that may need classification review, owner follow-up, or reporting support.

## 13. Ownership Type Classification

In [ ]:
#Classify Ownership Type for Buildings Missing Property Type

owner_col = "Property Owner Name"

#Fill missing owner names to avoid string errors
owner_series = missing_property[owner_col].fillna("").astype(str)

conditions = [

    #1 Property Name or Address-like
    owner_series.str.contains(
        r"^\d+\s+.*\b(?:STREET|ST|AVENUE|AVE|ROAD|RD|BLVD|LANE|LN|DRIVE|DR|COURT|CT|PLACE|PL)\b",
        case=False, na=False
    ),

    #2 Education
    owner_series.str.contains(
        r"SCHOOL|CHARTER|ACADEMY|COLLEGE|UNIVERSITY|EDUCATION",
        case=False, na=False
    ),

    #3 Public Safety
    owner_series.str.contains(
        r"POLICE|FIRE|EMS|PUBLIC SAFETY|PATROLMEN|DEPARTMENT",
        case=False, na=False
    ),

    #4 Condominium
    owner_series.str.contains(
        r"CONDO|CONDOMINIUM",
        case=False, na=False
    ),

    #5 Public Sector
    owner_series.str.contains(
        r"\bCITY\b|\bMASS\b|\bAUTHORITY\b|\bREDEVELOPMENT\b|\bMBTA\b|PUBLIC HEALTH|\bCOMM\b",
        case=False, na=False
    ),

    #6 Trust
    owner_series.str.contains(
        r"TRUST|TRST|\bTS\b",
        case=False, na=False
    ),

    #7 Association
    owner_series.str.contains(
        r"ASSOC|ASSOCIATION",
        case=False, na=False
    ),

    #8 Nonprofit or Religious
    owner_series.str.contains(
        r"CHURCH|FOUNDATION|SOCIETY|COMMUNITY|DIOCESE|METHODIST|BUDDHIST|CATHOLIC|FRIARS|SISTERS|HOLY|PARISH|ARCH|WORSHIP|WRSHP|TEMPLE|MINISTRY",
        case=False, na=False
    ),

    #9 Partnership
    owner_series.str.contains(
        r"\bLLP\b|\bLP\b|\bLPS\b|\bLIMITED PARTNERSHIP\b|\bPARTNERSHIP\b",
        case=False, na=False
    ),

    #10 LLC
    owner_series.str.contains(
        r"\bLLC\b",
        case=False, na=False
    ),

    #11 Corporate
    owner_series.str.contains(
        r"INC|CORP|COMPANY|HOLDINGS|REALTY|PROPERTIES",
        case=False, na=False
    ),

    #12 Business Venture
    owner_series.str.contains(
        r"\bGROUP\b|\bMGMT\b|\bMANAGEMENT\b|\bVENTURES\b|\bCAPITAL\b",
        case=False, na=False
    ),

    #13 Individual
    owner_series.str.contains(
        r"^[A-Za-z]+(?:\s+[A-Za-z\.]+){1,3}$",
        na=False
    )
]

choices = [
    "Property Name",
    "Education",
    "Public Safety",
    "Condominium",
    "Public Sector",
    "Trust",
    "Association",
    "Nonprofit",
    "Partnership",
    "LLC",
    "Corporate",
    "Business Venture",
    "Individual"
]

missing_property["ownership_type"] = np.select(
    conditions,
    choices,
    default="Other"
)

In [ ]:
#Ownership Type Summary

#Count ownership type classifications
ownership_counts = (
    missing_property["ownership_type"]
    .fillna("Unclassified")
    .value_counts()
)

#Calculate percentage distribution
ownership_percent = (
    missing_property["ownership_type"]
    .fillna("Unclassified")
    .value_counts(normalize=True)
    .mul(100)
    .round(2)
)

#Create summary table
ownership_summary = pd.DataFrame({
    "Ownership Type": ownership_counts.index,
    "Count": ownership_counts.values,
    "Percent (%)": ownership_percent.values
})

#Display styled table
display(
    ownership_summary.style
    .hide(axis="index")
    .set_caption("Ownership Type Distribution for Buildings Missing Property Type")
    .format({
        "Count": "{:,}",
        "Percent (%)": "{:.2f}%"
    })
    .set_properties(**{
        "text-align": "center",
        "padding": "12px",
        "font-size": "14px",
        "border": "1px solid #DADADA"
    })
    .set_table_styles([
        {
            "selector": "caption",
            "props": [
                ("caption-side", "top"),
                ("font-size", "18px"),
                ("font-weight", "bold"),
                ("padding", "10px")
            ]
        },
        {
            "selector": "th",
            "props": [
                ("text-align", "center"),
                ("font-size", "14px"),
                ("font-weight", "bold"),
                ("padding", "10px"),
                ("border", "1px solid #DADADA")
            ]
        }
    ])
)

### Observation

Missing property type information is concentrated among buildings classified through owner-name patterns as LLC-owned, public sector, and condominium-related properties. Together, these three categories account for 56.45% of buildings missing property type information. This pattern suggests that missing property type data may be connected to ownership structures that are more complex to identify or administer, including legal entities, shared governance models, and institutional ownership.

These categories should be interpreted as analytical approximations rather than official ownership designations. Because the classification depends on free-text owner names, some records may be misclassified or remain difficult to categorize. However, the relatively small share categorized as Other (3.81%) suggests that the keyword-based classification encapsulates most owner-name patterns in this subset.

In [ ]:
#Compliance Status by Ownership Type

#Create count table
compliance_counts = (
    missing_property
    .assign(ownership_type=missing_property["ownership_type"].fillna("Unclassified"))
    .groupby(["ownership_type", "compliance_status"])
    .size()
    .unstack(fill_value=0)
)

#Calculate row percentages
compliance_percent = compliance_counts.div(compliance_counts.sum(axis=1), axis=0) * 100

#Combine counts and percentages
compliance_summary = compliance_counts.copy()

for col in compliance_counts.columns:
    compliance_summary[col] = (
        compliance_counts[col].astype(str)
        + " ("
        + compliance_percent[col].round(1).astype(str)
        + "%)"
    )

#Reset index for display
compliance_summary = compliance_summary.reset_index()

#Display styled table
display(
    compliance_summary.style
    .hide(axis="index")
    .set_caption("Compliance Status by Ownership Type for Buildings Missing Property Type")
    .set_properties(**{
        "text-align": "center",
        "padding": "12px",
        "font-size": "14px",
        "border": "1px solid #DADADA"
    })
    .set_table_styles([
        {
            "selector": "caption",
            "props": [
                ("caption-side", "top"),
                ("font-size", "18px"),
                ("font-weight", "bold"),
                ("padding", "10px")
            ]
        },
        {
            "selector": "th",
            "props": [
                ("text-align", "center"),
                ("font-size", "14px"),
                ("font-weight", "bold"),
                ("padding", "10px"),
                ("border", "1px solid #DADADA")
            ]
        }
    ])
)

In [ ]:
#Overall Compliance Summary for Buildings Missing Property Type

#Use only numeric compliance-status columns
status_counts = compliance_counts.select_dtypes(include="number")

#Calculate total records across all compliance categories
total = status_counts.sum().sum()

#Summarize total counts by compliance category
summary = (
    status_counts
    .sum(axis=0)
    .reset_index()
)

summary.columns = ["Compliance Status", "Count"]

#Calculate percent of total
summary["Percent (%)"] = (summary["Count"] / total * 100).round(2)

#Display styled table
display(
    summary.style
    .hide(axis="index")
    .set_caption("Overall Compliance Summary for Buildings Missing Property Type")
    .format({
        "Count": "{:,.0f}",
        "Percent (%)": "{:.2f}%"
    })
)

### Observation

Among buildings missing property type information, 95.99% are marked “not submitted,” while 3.95% are listed as “state” and only 0.06% are pending revisions. This shows that missing property type is not spread evenly across compliance categories. Instead, it is concentrated among buildings that did not submit reporting information.

In [ ]:
#Ownership Type Distribution Visualization

#Calculate ownership type percentage distribution
ownership_counts = (
    missing_property["ownership_type"]
    .fillna("Unclassified")
    .value_counts(normalize=True)
    .mul(100)
    .round(2)
)

#Create intensity-based colors using normalized values
values = ownership_counts.values
normalized = values / values.max()

#Use matplotlib Blues colormap for intensity shading
colors = plt.cm.Blues(normalized)

#Create bar chart
plt.figure(figsize=(12, 6))

plt.bar(
    ownership_counts.index,
    ownership_counts.values,
    color=colors
)

plt.title(
    "Ownership Type Distribution for Buildings Missing Property Type",
    fontsize=14,
    fontweight="bold"
)

plt.xlabel(
    "Ownership Type",
    fontsize=12
)

plt.ylabel(
    "Percent of Buildings (%)",
    fontsize=12
)

plt.xticks(
    rotation=45,
    ha="right"
)

#Add percentage labels above bars
for i, value in enumerate(ownership_counts.values):
    plt.text(
        i,
        value + 0.2,
        f"{value:.1f}%",
        ha="center",
        fontsize=10
    )

plt.tight_layout()
plt.show()

### Observation

Buildings missing property type information are concentrated in a small number of inferred ownership categories. LLC-owned properties account for 20.2% of this group, followed by public sector entities at 18.5% and condominium-related properties at 17.7%. Together, these three categories represent 56.4% of buildings missing property type information.

This pattern suggests that missing property type data may be associated with ownership and administrative complexity, including legal entities, institutional ownership, and shared governance structures. These categories should be interpreted as analytical approximations based on owner-name patterns, not official ownership classifications.

## 14. Energy Performance by Compliance Status

In [ ]:
#Energy Performance by Compliance Status

#Create cleaned dataset using only buildings with valid Site EUI
df_eui_valid = (
    df_full
    .dropna(subset=["site_eui"])
    .copy()
)

#Summarize Site EUI by compliance status
eui_by_compliance = (
    df_eui_valid
    .groupby("compliance_status")
    .agg(
        Building_Count=("site_eui", "count"),
        Average_Site_EUI=("site_eui", "mean"),
        Median_Site_EUI=("site_eui", "median"),
        Standard_Deviation=("site_eui", "std")
    )
    .reset_index()
    .sort_values(
        by="Average_Site_EUI",
        ascending=False
    )
)

#Rename columns for presentation
eui_by_compliance = eui_by_compliance.rename(columns={
    "compliance_status": "Compliance Status",
    "Building_Count": "Building Count",
    "Average_Site_EUI": "Average Site EUI",
    "Median_Site_EUI": "Median Site EUI",
    "Standard_Deviation": "Standard Deviation"
})

#Display styled table
display(
    eui_by_compliance.style
    .hide(axis="index")
    .set_caption("Site EUI Summary by Compliance Status")
    .format({
        "Building Count": "{:,.0f}",
        "Average Site EUI": "{:.2f}",
        "Median Site EUI": "{:.2f}",
        "Standard Deviation": "{:.2f}"
    })
)

### Table Note

This analysis only includes buildings with valid Site EUI values. Buildings without reported Site EUI, including most non-submitted records, are excluded from this comparison.

### Observation

Among buildings with valid Site EUI data, state-listed buildings have the highest average Site EUI at 92.03 kBtu/ft², followed by buildings marked in compliance at 76.41 kBtu/ft² and buildings pending revisions at 73.99 kBtu/ft². However, the median values show a more nuanced pattern. State-listed buildings also have the highest median Site EUI at 76.10 kBtu/ft², while in-compliance buildings have a median of 63.80 kBtu/ft² and pending-revisions buildings have a much lower median of 39.65 kBtu/ft².

This comparison should be interpreted carefully because the compliance groups differ substantially in size. The state category includes only 34 buildings with valid Site EUI values, while the in-compliance category includes 3,412 buildings. The pending-revisions category also has a high standard deviation of 94.68, suggesting substantial variation or outliers. Overall, this table shows that compliance status and energy performance are related but not identical measures: a building may be listed as in compliance while still having relatively high energy use.

## 15. Energy Performance by Property Type

In [ ]:
#Summarize Site EUI by property type with sample size

property_eui_summary = (
    df_clean
    .groupby("property_type")
    .agg(
        Building_Count=("site_eui", "count"),
        Average_Site_EUI=("site_eui", "mean"),
        Median_Site_EUI=("site_eui", "median"),
        Standard_Deviation=("site_eui", "std")
    )
    .reset_index()
    .sort_values("Average_Site_EUI", ascending=False)
)

display(
    property_eui_summary.style
    .hide(axis="index")
    .set_caption("Site EUI Summary by Property Type")
    .format({
        "Building_Count": "{:,}",
        "Average_Site_EUI": "{:.1f}",
        "Median_Site_EUI": "{:.1f}",
        "Standard_Deviation": "{:.1f}"
    })
    .set_table_styles(TABLE_STYLES)
)

### Table Note

This analysis only includes buildings with valid property type and Site EUI values. Property types with very small sample sizes should be interpreted cautiously because one or two buildings can strongly affect the average.

### Observation

Energy intensity varies substantially by property type. Among property types with enough records for more reliable comparison, healthcare, laboratory, supermarket, and other specialized facilities show some of the highest Site EUI values. Urgent care and outpatient facilities average 253.7 kBtu/ft², hospitals average 236.2 kBtu/ft², laboratories average 233.3 kBtu/ft², and supermarkets average 227.0 kBtu/ft².

By contrast, larger and more common property types such as multifamily housing and office buildings have lower average Site EUI values, at 67.8 kBtu/ft² and 61.0 kBtu/ft², respectively. This suggests that energy performance is strongly shaped by building use, not just compliance status. Specialized buildings often have higher operational energy needs because of equipment, ventilation, refrigeration, or extended operating hours.

## 16. Building Size and Energy Intensity Check

In [ ]:
#Correlation Analysis: Building Size vs Energy Intensity

#Calculate correlation between building size and Site EUI
corr = (
    df_clean[[
        "gross_floor_area",
        "site_eui"
    ]]
    .corr()
    .round(3)
)

#rename columns for cleaner display
corr = corr.rename(
    index={
        "gross_floor_area": "Gross Floor Area",
        "site_eui": "Site EUI"
    },
    columns={
        "gross_floor_area": "Gross Floor Area",
        "site_eui": "Site EUI"
    }
)

#Display styled correlation matrix
display(
    corr.style
    .set_caption("Correlation Matrix: Gross Floor Area vs Site EUI")
    .set_properties(**{
        "text-align": "center",
        "padding": "12px",
        "font-size": "14px",
        "border": "1px solid #DADADA"
    })
    .set_table_styles([
        {
            "selector": "caption",
            "props": [
                ("caption-side", "top"),
                ("font-size", "18px"),
                ("font-weight", "bold"),
                ("padding", "10px")
            ]
        },
        {
            "selector": "th",
            "props": [
                ("text-align", "center"),
                ("font-size", "14px"),
                ("font-weight", "bold"),
                ("padding", "10px"),
                ("border", "1px solid #DADADA")
            ]
        }
    ])
)

### Observation

The correlation between gross floor area and Site EUI is minimal, suggesting that building size alone does not explain energy intensity. This supports the earlier property-type analysis: operational use, building systems, equipment, and occupancy patterns likely matter more than size alone.

## 17. Energy Performance by Property Type and Compliance Status

In [ ]:
#Average Site EUI by Property Type and Compliance Status

#Remove rows with missing values needed for comparison
df_plot = (
    df_clean
    .dropna(subset=[
        "property_type",
        "compliance_status",
        "site_eui"
    ])
    .copy()
)

#Focus on the top 5 most common property types
top_types = (
    df_plot["property_type"]
    .value_counts()
    .head(5)
    .index
)

df_plot = df_plot[df_plot["property_type"].isin(top_types)]

#Calculate average Site EUI and counts
grouped = (
    df_plot
    .groupby(["property_type", "compliance_status"])
    .agg(
        Average_Site_EUI=("site_eui", "mean"),
        Building_Count=("site_eui", "count")
    )
    .reset_index()
)

#Create pivot table for grouped bar chart
pivot = grouped.pivot(
    index="property_type",
    columns="compliance_status",
    values="Average_Site_EUI"
)

#Sort by overall mean across compliance groups
pivot["sort_value"] = pivot.mean(axis=1)
pivot = pivot.sort_values("sort_value", ascending=False).drop(columns="sort_value")

#Create grouped bar chart
ax = pivot.plot(
    kind="bar",
    figsize=(12, 7)
)

plt.title(
    "Average Site EUI by Property Type and Compliance Status",
    fontsize=14,
    fontweight="bold"
)

plt.xlabel("Property Type", fontsize=12)
plt.ylabel("Average Site EUI (kBtu/ft²)", fontsize=12)
plt.xticks(rotation=30, ha="right")
plt.legend(title="Compliance Status", bbox_to_anchor=(1.02, 1), loc="upper left")
plt.tight_layout()
plt.show()

## 18. Complexity Classification

### Method Note: Interpretive Complexity Categories

The complexity categories used below are analytical groupings created for this project. They are not official BERDO classifications.

These categories are intended to help compare broad building types by likely retrofit and operational complexity. For example, multifamily, residential, and mixed-use buildings are grouped as higher complexity because they may involve multiple occupants, owners, or decision-makers. Office, retail, and hotel properties are treated as moderate complexity, while specialized institutional or technical facilities are grouped separately because their energy use patterns may reflect more complex building systems.

These groupings should be interpreted as a screening tool, not as a final compliance determination. They are useful for identifying where additional outreach, technical assistance, or retrofit planning may be needed, but they should be reviewed alongside building-specific context.

In [ ]:
#Property Type Complexity Classification

#Classify property types into interpretive retrofit coordination groups
def classify_complexity(property_type):
    if pd.isna(property_type):
        return "Unknown"

    property_type = property_type.lower()

    #High coordination complexity
    if any(x in property_type for x in [
        "multifamily",
        "residential",
        "mixed"
    ]):
        return "High Coordination Complexity"

    #Moderate coordination complexity
    if any(x in property_type for x in [
        "office",
        "retail",
        "hotel"
    ]):
        return "Moderate Coordination Complexity"

    #Specialized systems complexity
    if any(x in property_type for x in [
        "university",
        "hospital",
        "laboratory"
    ]):
        return "Specialized Systems Complexity"

    #Default classification
    return "Moderate Coordination Complexity"

#Apply classification
df_clean["complexity"] = (
    df_clean["property_type"]
    .apply(classify_complexity)
)

## 19. Energy Intensity Classification

In [10]:
#Energy Intensity Classification

#Use median Site EUI as threshold for classification
eui_threshold = df_clean["site_eui"].median()

#Classify buildings into High EUI vs Low EUI
df_clean["eui_category"] = (
    df_clean["site_eui"]
    .apply(
        lambda x:
        "High EUI" if x > eui_threshold
        else "Low EUI"
    )
)

#Display threshold value
threshold_table = pd.DataFrame({
    "Metric": ["Median Site EUI Threshold"],
    "Value": [round(eui_threshold, 2)]
})

display(
    threshold_table.style
    .hide(axis="index")
    .set_caption("Site EUI Classification Threshold")
    .set_properties(**{
        "text-align": "center",
        "padding": "12px",
        "font-size": "14px",
        "border": "1px solid #DADADA"
    })
    .set_table_styles([
        {
            "selector": "caption",
            "props": [
                ("caption-side", "top"),
                ("font-size", "18px"),
                ("font-weight", "bold"),
                ("padding", "10px")
            ]
        },
        {
            "selector": "th",
            "props": [
                ("text-align", "center"),
                ("font-size", "14px"),
                ("font-weight", "bold"),
                ("padding", "10px"),
                ("border", "1px solid #DADADA")
            ]
        }
    ])
)


#Summarize EUI category distribution

eui_category_summary = (
    df_clean["eui_category"]
    .value_counts()
    .reset_index()
)

eui_category_summary.columns = [
    "EUI Category",
    "Building Count"
]

eui_category_summary["Percent (%)"] = (
    eui_category_summary["Building Count"]
    / eui_category_summary["Building Count"].sum()
    * 100
)

display(
    eui_category_summary.style
    .hide(axis="index")
    .set_caption("Building Count by EUI Category")
    .format({
        "Building Count": "{:,}",
        "Percent (%)": "{:.1f}%"
    })
    .set_table_styles(TABLE_STYLES)
)

Metric,Value
Median Site EUI Threshold,63.300000


EUI Category,Building Count,Percent (%)
Low EUI,"1,844",50.1%
High EUI,"1,834",49.9%


**Observation**

The median Site EUI threshold is 63.30 kBtu/ft². Buildings above this value are classified as High EUI, while buildings at or below this value are classified as Low EUI. The resulting groups are nearly evenly split, with 1,844 buildings classified as Low EUI (50.1%) and 1,834 buildings classified as High EUI (49.9%).

This classification is not an official BERDO performance standard. It is an analytical screening tool used to compare buildings with relatively higher and lower reported energy intensity within this dataset.

### Table 5. Selected BERDO 2.0 Emissions Performance Standards by Building Use

| Building use | 2025–2029 | 2030–2034 | 2035–2039 | 2040–2044 | 2045–2049 | 2050 |
|---|---:|---:|---:|---:|---:|---:|
| Multifamily housing | 4.1 | 2.4 | 1.8 | 1.1 | 0.6 | 0 |
| Office | 5.3 | 3.2 | 2.4 | 1.6 | 0.8 | 0 |
| Retail | 7.1 | 3.4 | 2.4 | 1.5 | 0.7 | 0 |
| Food Sales & Service | 17.4 | 10.9 | 8.0 | 5.4 | 2.7 | 0 |
| Healthcare | 15.4 | 10.0 | 7.4 | 4.9 | 2.4 | 0 |
| Lodging | 5.8 | 3.7 | 2.7 | 1.8 | 0.9 | 0 |
| Manufacturing / Industrial | 23.9 | 15.3 | 10.9 | 6.7 | 3.2 | 0 |
| College / University | 10.2 | 5.3 | 3.8 | 2.5 | 1.2 | 0 |
| Education | 3.9 | 2.4 | 1.8 | 1.2 | 0.6 | 0 |
| Technology / Science | 19.2 | 11.1 | 7.8 | 5.1 | 2.5 | 0 |
| Assembly | 7.8 | 4.6 | 3.3 | 2.1 | 1.1 | 0 |
| Services | 7.5 | 4.5 | 3.3 | 2.2 | 1.1 | 0 |
| Storage | 5.4 | 2.8 | 1.8 | 1.0 | 0.4 | 0 |

**Unit:** kg CO₂e / square foot / year  
**Source:** City of Boston, BERDO 2.0 CO₂e Emissions Standards by Building Use.

### Method Note

The BERDO standards above are included for policy context. BERDO emissions performance is measured in kg CO₂e per square foot per year, while this analysis uses Site EUI in kBtu/ft² as a screening metric. Therefore, the High EUI and Low EUI categories in this notebook should not be interpreted as official BERDO compliance determinations.

## 20. Estimated GHG Emissions Intensity Analysis

In [13]:
#Estimated GHG Emissions Intensity Analysis

df_clean["ghg_intensity_kgco2e_sqft"] = (
    df_clean["ghg_emissions"] / df_clean["gross_floor_area"]
)

ghg_intensity_summary = pd.DataFrame({
    "Metric": [
        "Average estimated GHG intensity",
        "Median estimated GHG intensity",
        "Minimum estimated GHG intensity",
        "Maximum estimated GHG intensity"
    ],
    "Value": [
        df_clean["ghg_intensity_kgco2e_sqft"].mean(),
        df_clean["ghg_intensity_kgco2e_sqft"].median(),
        df_clean["ghg_intensity_kgco2e_sqft"].min(),
        df_clean["ghg_intensity_kgco2e_sqft"].max()
    ]
})

display(
    style_table(
        ghg_intensity_summary.style,
        "Estimated GHG Emissions Intensity Summary",
        {
            "Value": "{:.2f}"
        }
    )
)

Metric,Value
Average estimated GHG intensity,4.25
Median estimated GHG intensity,3.52
Minimum estimated GHG intensity,-0.03
Maximum estimated GHG intensity,35.50


In [ ]:
#Estimated GHG Emissions Intensity by Property Type

ghg_by_property_type = (
    df_clean
    .dropna(subset=[
        "property_type",
        "ghg_intensity_kgco2e_sqft"
    ])
    .groupby("property_type")
    .agg(
        Building_Count=("ghg_intensity_kgco2e_sqft", "count"),
        Average_GHG_Intensity=("ghg_intensity_kgco2e_sqft", "mean"),
        Median_GHG_Intensity=("ghg_intensity_kgco2e_sqft", "median")
    )
    .reset_index()
    .sort_values("Average_GHG_Intensity", ascending=False)
)

display(
    style_table(
        ghg_by_property_type.style,
        "Estimated GHG Emissions Intensity by Property Type",
        {
            "Building_Count": "{:,}",
            "Average_GHG_Intensity": "{:.2f}",
            "Median_GHG_Intensity": "{:.2f}"
        }
    )
)

### Observation

This section moves the analysis closer to BERDO’s official emissions performance framework by calculating estimated greenhouse gas emissions intensity in kg CO₂e per square foot. Unlike Site EUI, which measures energy use, this metric reflects estimated emissions relative to building size.

The property type comparison shows which building categories have higher estimated emissions intensity, not just higher energy use intensity. This distinction matters because two buildings can have similar Site EUI values but different emissions profiles depending on fuel mix, electricity use, and building operations.

This analysis should still be interpreted as exploratory rather than a final BERDO compliance determination. A complete BERDO compliance test would require matching each property type to the correct BERDO building use category, applying any blended standard for mixed use buildings, and confirming the relevant compliance year.

## 21. Priority Matrix

In [ ]:
#Priority Matrix

#Create priority column using updated complexity labels
conditions = [
    (df_clean["site_eui"] >= eui_threshold) & (df_clean["complexity"] == "High Coordination Complexity"),
    (df_clean["site_eui"] >= eui_threshold) & (df_clean["complexity"] == "Moderate Coordination Complexity"),
    (df_clean["site_eui"] >= eui_threshold) & (df_clean["complexity"] == "Specialized Systems Complexity"),
    (df_clean["site_eui"] < eui_threshold) & (df_clean["complexity"] == "High Coordination Complexity"),
    (df_clean["site_eui"] < eui_threshold) & (df_clean["complexity"] == "Moderate Coordination Complexity"),
    (df_clean["site_eui"] < eui_threshold) & (df_clean["complexity"] == "Specialized Systems Complexity")
]

choices = [
    "High Priority",
    "Medium Priority",
    "Medium Priority",
    "Medium Priority",
    "Low Priority",
    "Low Priority"
]

df_clean["priority"] = np.select(
    conditions,
    choices,
    default="Unclassified"
)

#Numeric mapping for plotting
complexity_map = {
    "Unknown": 0,
    "Moderate Coordination Complexity": 1,
    "High Coordination Complexity": 2,
    "Specialized Systems Complexity": 3
}

#Summarize priority distribution

priority_summary = (
    df_clean["priority"]
    .value_counts()
    .reset_index()
)

priority_summary.columns = [
    "Priority Category",
    "Building Count"
]

priority_summary["Percent (%)"] = (
    priority_summary["Building Count"]
    / priority_summary["Building Count"].sum()
    * 100
)

display(
    priority_summary.style
    .hide(axis="index")
    .set_caption("Building Priority Distribution")
    .format({
        "Building Count": "{:,}",
        "Percent (%)": "{:.1f}%"
    })
    .set_table_styles(TABLE_STYLES)
)

### Observation

The priority matrix classifies buildings into broad screening groups based on reported Site EUI and interpretive complexity categories. Most buildings fall into the Medium Priority group, with 1,893 buildings or 51.5% of the classified dataset. The High Priority group includes 1,007 buildings, representing 27.4% of buildings, while 777 buildings or 21.1% are classified as Low Priority.

The High Priority group is especially important for planning because it combines above median Site EUI with high coordination complexity. These buildings may warrant closer review, targeted outreach, technical assistance, or retrofit planning support.

This classification is not an official BERDO ranking. It should be read alongside the estimated GHG emissions intensity analysis above, which is closer to BERDO’s official kg CO₂e per square foot framework. Together, the two metrics help distinguish energy use screening from emissions performance context.

In [ ]:
#High Priority Building Summary

#Count buildings classified as High Priority
high_priority_count = (
    df_clean[df_clean["priority"] == "High Priority"]
    .shape[0]
)

#Total number of buildings in cleaned dataset
total_buildings = df_clean.shape[0]

#Calculate percentage of total dataset
percent = (
    high_priority_count
    / total_buildings
    * 100
)

#Create summary table
priority_summary = pd.DataFrame({
    "Metric": [
        "High Priority Buildings",
        "Total Buildings Analyzed",
        "Percent of Dataset"
    ],
    "Value": [
        f"{high_priority_count:,}",
        f"{total_buildings:,}",
        f"{percent:.2f}%"
    ]
})

display(
    priority_summary.style
    .hide(axis="index")
    .set_caption("High Priority Building Summary")
    .set_table_styles(TABLE_STYLES)
)

In [ ]:
#Compliance Status of High Priority Buildings

#Filter high priority buildings using priority label
high_priority_buildings = (
    df_clean[df_clean["priority"] == "High Priority"]
    .copy()
)

#Count compliance status distribution
high_priority_compliance = (
    high_priority_buildings["compliance_status"]
    .value_counts()
    .reset_index()
)

high_priority_compliance.columns = [
    "Compliance Status",
    "Building Count"
]

#Calculate percentage distribution
high_priority_compliance["Percent (%)"] = (
    high_priority_compliance["Building Count"]
    / high_priority_compliance["Building Count"].sum()
    * 100
)

#Display styled table
display(
    high_priority_compliance.style
    .hide(axis="index")
    .set_caption("Compliance Status Distribution of High Priority Buildings")
    .format({
        "Building Count": "{:,}",
        "Percent (%)": "{:.2f}%"
    })
    .set_table_styles(TABLE_STYLES)
)

In [ ]:
#Outlier Buildings with Very High Energy Use

#Identify buildings with extremely high Site EUI
#Threshold used: Site EUI > 200

outliers_table = (
    df_clean[
        df_clean["site_eui"] > 200
    ][[
        "Building Address",
        "property_type",
        "gross_floor_area",
        "site_eui",
        "compliance_status"
    ]]
    .sort_values(
        by="site_eui",
        ascending=False
    )
    .reset_index(drop=True)
)

#Rename columns for cleaner presentation
outliers_table = (
    outliers_table.rename(columns={
        "Building Address": "Address",
        "property_type": "Property Type",
        "gross_floor_area": "Gross Floor Area",
        "site_eui": "Site EUI",
        "compliance_status": "Compliance Status"
    })
)

#Display top 20 highest EUI buildings
display(
    outliers_table
    .head(20)
    .style
    .hide(axis="index")
    .set_caption("Buildings with Very High Site EUI Requiring Further Review")
    .format({
        "Gross Floor Area": "{:,.0f}",
        "Site EUI": "{:.2f}"
    })
    .set_properties(**{
        "text-align": "center",
        "padding": "10px",
        "font-size": "13px",
        "border": "1px solid #DADADA"
    })
    .set_table_styles([
        {
            "selector": "caption",
            "props": [
                ("caption-side", "top"),
                ("font-size", "18px"),
                ("font-weight", "bold"),
                ("padding", "10px")
            ]
        },
        {
            "selector": "th",
            "props": [
                ("text-align", "center"),
                ("font-weight", "bold"),
                ("padding", "10px"),
                ("border", "1px solid #DADADA")
            ]
        }
    ])
)

**Observation**

Most of the highest EUI buildings exhibit extremely high energy use intensity (EUI > 200), yet are still classified as “In Compliance,” indicating that compliance status is not solely determined by energy intensity. Notably, some of these outliers are relatively small buildings, suggesting potential data anomalies or atypical usage patterns. 

The presence of high-energy outliers within the “Other” category reinforces the complexity and heterogeneity of this group, which may obscure important variations in building performance.

## 22. BERDO Intervention Type Summary

This section translates the analysis into practical intervention categories. Buildings with missing property type or missing Site EUI are classified as needing reporting support. Buildings with valid data and above-median Site EUI are classified as needing retrofit planning. Buildings with complete data and lower Site EUI are classified as monitor.

In [ ]:
#Classify buildings by likely intervention need

def classify_intervention(row):
    if pd.isna(row["property_type"]) or pd.isna(row["site_eui"]):
        return "Reporting Support"
    elif row["site_eui"] >= eui_threshold:
        return "Retrofit Planning / Technical Assistance"
    else:
        return "Monitor"

df_full["intervention_type"] = df_full.apply(classify_intervention, axis=1)

#Summarize intervention groups
intervention_summary = (
    df_full["intervention_type"]
    .value_counts()
    .reset_index()
)

intervention_summary.columns = [
    "Intervention Type",
    "Building Count"
]

intervention_summary["Percent of Buildings"] = (
    intervention_summary["Building Count"]
    / intervention_summary["Building Count"].sum()
)

display(
    intervention_summary.style
    .hide(axis="index")
    .set_caption("BERDO Intervention Type Summary")
    .format({
        "Building Count": "{:,}",
        "Percent of Buildings": "{:.1%}"
    })
    .set_table_styles(TABLE_STYLES)
)

## 23. Implementation Considerations

This analysis suggests that BERDO support can be organized around two broad needs: reporting support for buildings with incomplete or missing data, and retrofit planning support for buildings with reported data that suggests high energy intensity or future compliance risk. The ideas below are not formal policy recommendations. They are practical considerations based on the patterns identified in the dataset.

| Focus Area | What the Data Suggests | Possible Support Strategy | Why It Matters |
|---|---|---|---|
| Reporting completeness | Buildings marked Not Submitted or missing property type data may be difficult to evaluate for emissions performance or retrofit needs. | Provide clearer reporting guidance, technical assistance, and targeted outreach for buildings with incomplete information. | Complete reporting is the first step toward understanding compliance risk and planning future support. |
| Fragmented ownership | Ownership-name patterns suggest that some buildings may have complex or unclear responsibility structures. | Develop outreach materials that account for condominiums, LLC-managed properties, trusts, and other ownership structures. | These buildings may face administrative or coordination barriers rather than purely technical barriers. |
| High-EUI screening | Some reported buildings show high Site EUI, especially specialized uses such as healthcare, laboratories, food service, grocery stores, and hotels. | Use high-EUI buildings as candidates for deeper review, energy audits, or operational assessments. | High EUI does not prove noncompliance, but it can help identify buildings where further review may be useful. |
| Resource navigation | Some owners may need help understanding available incentives, financing options, audits, or retrofit pathways. | Connect reporting and benchmarking insights with existing technical assistance and incentive programs. | BERDO implementation may require long-term planning, not just annual reporting. |
| Ongoing monitoring | Compliance status, property type, EUI, and ownership indicators can change over time. | Track reporting gaps and high-EUI patterns across future BERDO datasets. | Repeated analysis can show whether support is reaching the buildings most likely to need it. |

### Summary

The main action implied by this analysis is not to treat all buildings the same. Buildings with missing or incomplete reporting data may need help with basic compliance steps, while buildings with complete data but high energy intensity may need planning support, technical review, or connections to existing retrofit resources.

This framework should be interpreted as a starting point for discussion rather than a final implementation plan. More detailed recommendations would require additional information, including owner capacity, building systems, capital plans, tenant considerations, and available program resources.

## 24. Conclusion

This analysis shows that BERDO compliance should be understood as both a reporting challenge and a long-term implementation challenge. The data points to two broad support needs: helping buildings complete reporting requirements and helping high-energy-use buildings plan for future performance improvements.

The first challenge is data completeness. A meaningful share of buildings did not submit required reporting information, and missing property classification appears closely connected to this issue. Buildings without an assigned property type are not evenly distributed across the dataset. They are concentrated in incomplete reporting categories, especially “Not Submitted.” This suggests that missing data is more than a technical limitation. It can also function as an early indicator of compliance risk and reduced policy visibility.

These reporting gaps may be connected to ownership and coordination challenges. Buildings associated with fragmented ownership structures, including condominiums, LLC-managed properties, and some public or institutional properties, may face additional barriers in identifying who is responsible for reporting, accessing technical support, and completing compliance requirements.

The second challenge is operational performance. Among buildings that do report, Site Energy Use Intensity varies substantially by property type. Specialized and energy-intensive uses, including healthcare facilities, laboratories, supermarkets, food service, restaurants, hotels, and some industrial buildings, appear among the higher-EUI categories. These results should be interpreted cautiously because some property-type groups have small sample sizes. Still, the broader pattern suggests that certain building functions may involve deeper operational complexity and longer-term retrofit needs.

This creates two broad support needs. The first involves buildings with incomplete or missing reporting information, which may benefit from clearer guidance, outreach, reporting assistance, and ownership coordination support. The second involves buildings with complete data but higher energy intensity, which may require deeper technical review, retrofit planning, financing guidance, and decarbonization strategies. Treating both groups the same would overlook the different barriers they face.

The most important finding is that compliance outcomes and data availability are not evenly distributed across Boston’s building stock. Larger and more standardized property types, such as multifamily housing and office buildings, generally show stronger reporting consistency, while buildings with missing classifications and broad “Other” categories contribute more heavily to reporting gaps and reduced visibility.

Meaningful emissions reductions under BERDO will require more than identifying high-energy buildings. It will also require improving data completeness, reducing reporting friction, and supporting ownership groups that may have limited capacity to engage with compliance systems. Using missing data as an early warning signal can help shift BERDO analysis from simply measuring noncompliance toward identifying where support may be needed most.

The central takeaway is that effective BERDO implementation depends on both energy performance and data visibility. A building’s ability to report, classify itself accurately, and navigate ownership responsibilities shapes how visible it is within the policy system and how effectively it can be supported over time.

## 25. Reflection: What I Learned

The concentration of missing property type data among “Not Submitted” buildings suggests that missing data is not simply a random data-quality issue. Through both quantitative analysis and stakeholder interviews, I came to understand that missing data can signal broader ownership and coordination challenges rather than simple reporting mistakes.

Smaller and more fragmented ownership groups, particularly condominiums, LLC-managed properties, trusts, and certain public or institutional buildings, may face difficulty identifying who is responsible for compliance, securing access to engineering expertise, and managing the time-intensive reporting process. Many may also lack dedicated facilities staff, long-term capital planning systems, or clear decision-making structures, which can make both reporting and retrofit planning more difficult.

This helped shift my understanding of compliance from a purely technical issue to an operational and organizational one. Buildings that fail to report may often be the same buildings that lack the internal capacity to engage with BERDO requirements in the first place.

As a result, missing property type data should not be treated simply as a limitation in the dataset. It can function as a measurable warning signal for compliance risk and can help identify where policy support, technical assistance, and targeted outreach may be most needed. This reinforced the importance of using data not only to measure outcomes, but also to understand where the system itself may create barriers to participation.

By adding estimated GHG emissions intensity alongside Site EUI, the analysis distinguishes between energy-use screening and emissions-performance context, while avoiding a final compliance determination without official building-use mapping and blended-standard calculations.